barretenberg/cpp/src/barretenberg/flavor/ultra.hpp

```diff
@@ -307,6 +407,28 @@ class UltraFlavor {
      */
     class ProverPolynomials : public AllEntities<Polynomial> {
       public:
+        ProverPolynomials(ProvingKey* proving_key)
```

Comment:
maybe should just be a reference here (maybe even const?)

Reply:
yeah esp once ProvingKey is not handled through a shared pointer